src/models/BonusNumber.js

```diff
@@ -0,0 +1,58 @@
+import LOTTO_CONSTANTS from '../constants/lottoConstatns.js';
+import ERROR_MESSAGES from '../constants/messages/errorMessages.js';
+
+class BounsNumber {
+  #winningNumbers;
+  #bonusNumber;
+
+  constructor(winningNumbers, bonusNumber) {
+    this.#winningNumbers = winningNumbers;
+    this.#validateBonusNumber(bonusNumber);
+    this.#bonusNumber = Number(bonusNumber);
+  }
+
+  get bonusNumber() {
+    return this.#bonusNumber;
+  }
+
+  #validateBonusNumber(bonusNumber) {
+    this.#validateType(bonusNumber);
+    this.#validateDuplication(bonusNumber);
+    this.#validateRange(bonusNumber);
+    this.#validateInteger(bonusNumber);
+  }
+
+  #validateType(bonusNumber) {
+    if (Number.isNaN(Number(bonusNumber))) {
+      throw new Error(
+        `${ERROR_MESSAGES.ERROR_PREFIX}${ERROR_MESSAGES.BONUS_NUMBER_TYPE_NOT_NUMBER}`,
+      );
+    }
+  }
+
+  #validateDuplication(bonusNumber) {
+    if (this.#winningNumbers.includes(Number(bonusNumber))) {
+      throw new Error(`${ERROR_MESSAGES.ERROR_PREFIX}${ERROR_MESSAGES.BONUS_NUMBER_DUPLICATION}`);
+    }
+  }
+
+  #validateRange(bonusNumber) {
+    const isValid =
+      Number(bonusNumber) > LOTTO_CONSTANTS.END_NUMBER ||
+      Number(bonusNumber) < LOTTO_CONSTANTS.START_NUMBER;
+
+    if (isValid) {
+      throw new Error(
+        `${ERROR_MESSAGES.ERROR_PREFIX}${ERROR_MESSAGES.BONUS_NUMBER_RANGE_NOT_RIGHT}`,
+      );
+    }
+  }
```

Comment:
```suggestion
  #validateRange(bonusNumber) {
    if (Number(bonusNumber) > LOTTO_CONSTANTS.END_NUMBER || Number(bonusNumber) < LOTTO_CONSTANTS.START_NUMBER) {
      throw new Error(
        `${ERROR_MESSAGES.ERROR_PREFIX}${ERROR_MESSAGES.BONUS_NUMBER_RANGE_NOT_RIGHT}`,
      );
    }
  }
```

다른 함수들처럼 인라인으로 작성하는것도 좋을거 같아요!

Reply:
맞습니다! 다만 이렇게 뺀 이유는 해당 메소드만 조건이 너무 긴 것 같아서 따로 빼게 되었는데요... 통일성을 부여하는게 더 좋을수도 있겠다라는 생각이 드네요. 짚어주셔서 감사합니다!